dependency.go

```diff
@@ -0,0 +1,171 @@
+package testcontainers
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types/container"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"golang.org/x/exp/slices"
+)
+
+// ContainerDependency represents a reliance that a container has on another container.
+type ContainerDependency struct {
+	Request ContainerRequest
+	EnvKey  string
```

Comment:
The environment variable can be injected by the callback function and not all containers can take advantage of it, so I'm not convinced this should be implemented in the dependency struct as a default behavior.

Reply:
Sorry, i'm not sure what you mean by the env variable can be injected by the callback func. The container would be already running by the time the callback function is called and I'm not aware if injecting env vars into a live container would be possible at that point.

Reply:
Do you mean that we should make Env injection optional through the use of something like `WithEnvName(envKey string)` ?

Reply:
Is there any reason why the env would be a requirement?